test/hotspot/jtreg/compiler/loopopts/superword/TestDependencyOffsets.java

```diff
@@ -486,15418 +409,475 @@
-        testDoubleP63(goldDoubleP63);
-        init(goldDoubleM64);
-        testDoubleM64(goldDoubleM64);
-        init(goldDoubleP64);
-        testDoubleP64(goldDoubleP64);
-        init(goldDoubleM65);
-        testDoubleM65(goldDoubleM65);
-        init(goldDoubleP65);
-        testDoubleP65(goldDoubleP65);
-        init(goldDoubleM128);
-        testDoubleM128(goldDoubleM128);
-        init(goldDoubleP128);
-        testDoubleP128(goldDoubleP128);
-        init(goldDoubleM129);
-        testDoubleM129(goldDoubleM129);
-        init(goldDoubleP129);
-        testDoubleP129(goldDoubleP129);
-        init(goldDoubleM192);
-        testDoubleM192(goldDoubleM192);
-        init(goldDoubleP192);
-        testDoubleP192(goldDoubleP192);
+/*
+ * We want to test SuperWord / AutoVectorization with different constant offsets (positive and negative):
+ *   for (int i = ...) { a[i + offset] = b[i] * 11; }
+ *
+ * To test aliasing, we have 3 modes: single-array, aliasing and non-aliasing.
+ * We test for various primitive types (int, long, short, char, byte, float, double).
+ * We run all test under various settings of MaxVectorSize and +-AlignVector.
+ * Finally, we verify the results and check that vectors of the expected length were created (IR rules).
+ */
+public class TestDependencyOffsets {
+    private static final Random RANDOM = Utils.getRandomInstance();
+    private static final int SIZE = 5_000 + RANDOM.nextInt(1000);
+
+    /*
+     * Template for the inner test class.
+     */
+    private static String generate(CompileFramework comp, String[] flags) {
+        return String.format("""
+               import compiler.lib.ir_framework.*;
+
+               public class InnerTest {
+                   private static int SIZE = %s;
+
+                   public static void main(String args[]) {
+                       TestFramework framework = new TestFramework(InnerTest.class);
+                       framework.addFlags("-classpath", "%s");
+                       framework.addFlags(%s);
+                       framework.setDefaultWarmup(0);
+                       framework.start();
+                   }
+
+                   // ------------------------- Init ---------------------------
+               %s
+
+                   // ------------------------- Verify -------------------------
+               %s
+
+                   // ------------------------- Tests --------------------------
+               %s
+               }
+               """,
+               SIZE,
+               comp.getEscapedClassPathOfCompiledClasses(),
+               Arrays.stream(flags).map(s -> "\"" + s + "\"").collect(Collectors.joining(", ")),
+               Arrays.stream(types).map(type -> type.generateInit()).collect(Collectors.joining("\n")),
+               Arrays.stream(types).map(type -> type.generateVerify()).collect(Collectors.joining("\n")),
+               getTests().stream().map(test -> test.generate()).collect(Collectors.joining("\n")));
```

Comment:
Since you don't need the lambda parameter, you can directly use a method reference:
```suggestion
               Arrays.stream(types).map(Type::generateInit).collect(Collectors.joining("\n")),
               Arrays.stream(types).map(Type::generateVerify).collect(Collectors.joining("\n")),
               getTests().stream().map(TestDefinition::generate).collect(Collectors.joining("\n")));
```